### PR TITLE
1) in case when appdomain is created by testhost (when disableappdoma…

### PR DESCRIPTION
**Pull request by acesiddhu**
## Description
1) in case when appdomain is created by testhost (when disableappdomin is set to true) we also try to unload that domain. we have multiple number of issues related to appdomain unload like crashes and hangs.

2) Will publish an RFC for testadapter writers and customer on how to workaround issues cropping up with appdomains. this change wont go before publishing RFC
3) added test which check if config file get merged in case appdomain is disabled.

RFC https://github.com/Microsoft/vstest-docs/pull/123

### CODE REVIEW

**Comment by acesiddhu** [comment]
@dotnet-bot test this please

**Review by cltshivash — APPROVED** [review]
:shipit:

File: TestPlatform.sln [hunk 1]
```diff
@@ -159,6 +159,10 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "perf", "perf", "{0D4DF78D-7
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "vstest.console.PlatformTests", "test\vstest.console.PlatformTests\vstest.console.PlatformTests.csproj", "{8C068694-23A2-47A2-A0DD-DB82D0AF0142}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "DisableAppdomainTest1", "test\TestAssets\DisableAppdomainTest1\DisableAppdomainTest1.csproj", "{CC7072C5-4CD0-4D33-97D0-29DF765AFFC2}"
```
Comment (smadala) [inline TestPlatform.sln]: Add TestAssets projects to https://github.com/Microsoft/vstest/tree/master/test/TestAssets/TestAssets.sln

File: test/Microsoft.TestPlatform.AcceptanceTests/DisableAppdomainTests.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+namespace Microsoft.TestPlatform.AcceptanceTests
+{
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+
+    [TestClass]
+    public class DisableAppdomainTests : AcceptanceTestBase
```
Comment (smadala) [inline test/Microsoft.TestPlatform.AcceptanceTests/DisableAppdomainTests.cs]: We already have test for this vstest/test/Microsoft.TestPlatform.AcceptanceTests/AppDomainTests.cs

Comment (acesiddhu) [reply test/Microsoft.TestPlatform.AcceptanceTests/DisableAppdomainTests.cs]: synced up offline. New test is doing more stuff. so retaining it

File: TestPlatform.sln (hunk 1 above)
Comment (acesiddhu) [reply TestPlatform.sln]: done

**Review by smadala — APPROVED** [review]

**Comment by acesiddhu** [comment]
@dotnet-bot test Windows_NT / Release Build

**Pull request merged**